src/types.ts

```diff
@@ -831,3 +831,24 @@ export interface CheckFormInvalidSubmissionEvent {
   errs: FieldErrors<CheckFormValues>;
   source: string;
 }
+
+export type PluginPermissions =
+  | 'grafana-synthetic-monitoring-app:read'
+  | 'grafana-synthetic-monitoring-app:write'
+  | 'grafana-synthetic-monitoring-app.checks:read'
+  | 'grafana-synthetic-monitoring-app.checks:write'
+  | 'grafana-synthetic-monitoring-app.checks:delete'
+  | 'grafana-synthetic-monitoring-app.probes:read'
+  | 'grafana-synthetic-monitoring-app.probes:write'
+  | 'grafana-synthetic-monitoring-app.probes:delete'
+  | 'grafana-synthetic-monitoring-app.alerts:read'
+  | 'grafana-synthetic-monitoring-app.alerts:write'
+  | 'grafana-synthetic-monitoring-app.alerts:delete'
+  | 'grafana-synthetic-monitoring-app.thresholds:read'
+  | 'grafana-synthetic-monitoring-app.thresholds:write'
+  | 'grafana-synthetic-monitoring-app.access-tokens:read'
+  | 'grafana-synthetic-monitoring-app.access-tokens:write'
+  | 'grafana-synthetic-monitoring-app.access-tokens:delete'
+  | 'grafana-synthetic-monitoring-app.plugin:enable'
+  | 'grafana-synthetic-monitoring-app.plugin:disable'
+  | 'grafana-synthetic-monitoring-app.tokens:delete';
```

Comment:
I think this could be a great contender for template literal types
Note: I'm not a TS Jedi, so there are probably better ways to do it than my examples.

Two examples, depending on how strict it needs to be
```
type PermissionBase = 'grafana-synthetic-monitoring-app';
type PermissionAction = 'read' | 'write' | 'delete' | 'enable' | 'disable';
type PermissionSubset = 'checks' | 'probes' | 'alerts' | 'thresholds' | 'access-tokens' | 'tokens' | 'plugin';
export type PluginPermissions =
  | `${PermissionBase}:${PermissionAction}`
  | `${PermissionBase}:${PermissionSubset}:${PermissionAction}`;
```

A bit stricter type
```
type PermissionBase = 'grafana-synthetic-monitoring-app';
type PluginPermissions =
  | `${PermissionBase}:${'read' | 'write'}`
  | `${PermissionBase}.checks:${'read' | 'write' | 'delete'}`
  | `${PermissionBase}.probes:${'read' | 'write' | 'delete'}`
  | `${PermissionBase}.alerts:${'read' | 'write' | 'delete'}`
  | `${PermissionBase}.thresholds:${'read' | 'write'}`
  | `${PermissionBase}.access-tokens:${'read' | 'write' | 'delete'}`
  | `${PermissionBase}.plugin:${'enable' | 'disable'}`
  | `${PermissionBase}.tokens:${'delete'}`;
```

Also, it feels like `enable|disable` are redundant (wouldn't `write|delete` be used instead).